### PR TITLE
MODINREACH-75 Central server Configuration POST 400 error

### PR DESCRIPTION
**Pull request by maryna-steshenko**
## Purpose

Fix Central server Configuration POST 400 error.
## Approach

- Handling of DataIntegrityViolationException in ExceptionHandlerController modified.
- OpenAPI configuration modified.
- CentralServerController test added.

## Pre-Merge Checklist:
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [ ] Duplications on new code is 3% or less
  - [ ] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [ ] There are no breaking changes in this PR.
  
If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?  
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.  

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/main/java/org/folio/innreach/controller/CentralServerController.java [hunk 1]
```diff
@@ -48,15 +50,20 @@
...
   }
 
   @Override
   @PostMapping
   public ResponseEntity<CentralServerDTO> postCentralServer(@Valid CentralServerDTO centralServerDTO) {
-    var createdCentralServer = centralServerService.createCentralServer(centralServerDTO);
+    CentralServerDTO createdCentralServer;
+    try {
+      createdCentralServer = centralServerService.createCentralServer(centralServerDTO);
+    } catch (DataIntegrityViolationException e) {
```
Comment (dmtkachenko) [inline src/main/java/org/folio/innreach/controller/CentralServerController.java]: I'd probably place this catch inside service method (`createCentralServer`)

**Comment by dmtkachenko** [comment]
I general looks good

**Comment by oliinyko** [comment]
@maryna-steshenko @dmtkachenko 
I would suggest modifying existing exception handler for `DataIntegrityViolationException` instead of creating a custom one. Otherwise we will be forced to add similar try/catch to every post/put where the violation may occur. By the way, PUT central server needs to be covered too as this issue reproduces during update.

The generic solution would be:
```
  @ExceptionHandler(DataIntegrityViolationException.class)
  @ResponseStatus(HttpStatus.CONFLICT)
  public Error handleDataIntegrityViolationException(DataIntegrityViolationException e) {
    return createError(HttpStatus.CONFLICT, e.getMessage());
  }
```
This would cover both post and put

**Review by oliinyko — APPROVED** [review]
Looks good to me

**Review by dmtkachenko — APPROVED** [review]

**Pull request merged**